Dynatrace/dynatrace-operator
# Use explicit tenant UUID returned by tenant

## Description

Currently the tenant UUID is derived from the URL provided in the dynakube.
It is never guaranteed that the url contains a valid UUID and can even be an alias. 

This PR updates the corresponding code to use the UUID returned by the tenant itself.

[Jira Ticket](https://dt-rnd.atlassian.net/browse/K8S-9537)


## How can this be tested?

Run cloudNative e2e test and check if everything works as expected. 
Run resilience test to verify that even with limited connectivity the CSI driver works as expected.
